pandas/_libs/lib.pyx

```diff
@@ -2357,3 +2354,115 @@ def fast_multiget(dict mapping, ndarray keys, default=np.nan):
             output[i] = default
 
     return maybe_convert_objects(output)
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef inline void convert_and_set_item(object item, Py_ssize_t index,
+                                      object[:] result,
+                                      bint keep_trivial_numbers):
+    cdef:
+        bint do_convert = 1
+        float64_t float_item
+
+    if keep_trivial_numbers:
+        if isinstance(item, int):
+            if <int>item == 0:
+                do_convert = 0
+        elif isinstance(item, float):
+            float_item = item
+            if float_item == 0.0 or float_item != float_item:
+                do_convert = 0
+
+    if do_convert and not isinstance(item, (str, bytes)):
```

Comment:
why would this *ever* be bytes? this is not typical to do this as very error prone (or rather is this every *not* bytes), IOW who is calling this?

Reply:
We do not know exactly in what format these strings come, because it is called in many places. So, we handle both cases, assuming that bytes format is valid case, and thus avoid of excess conversion from bytes to unicode. Is this wrong approach?
Why might the string representation in bytes format be incorrect?

Reply:
Never mind. Just changed to check only `str`